### DISCUSSION TITLE
Air SDK vs Air SDK For Flex Strange Issue [Flash Develop]

### DISCUSSION THREAD

**Discussion by dmunsie** [airsdk/Adobe-Runtime-Support#3124] [General]
When using either SDK I can build APKs no problem. The problem is compiling the .swf file. Using the Air SDK for Flex I have no problems. But when I try to compile the swf using the Air SDK w/ new compiler I get a "Type was not found or was not a compile-time constant" error on some of my custom classes.

One particular class has exactly 3999 lines of code. Is there a limit to the number of lines of code a class can have? I did some experimenting with removing a lot of lines of code and sure enough, somewhere around 1800-2000 lines of code in the class everything compiled fine.

But when I added back to the project another class that has only 1375 lines of code, that particular class is causing the error to show up again.  So is there any differences between the 2 different compilers I should be aware of?

My goal is to use the **[inline]** feature available when using the Air SDK with the updated compiler to help speed up a few routines. Does it speed things up enough to make it worth while to pursue a fix for this problem or should I just stick to using the Air SDK For Flex compiler? Thx for any comments.

**Comment by ajwfrost** [comment]
Hi

I know that AS2 had a limitation with the size of a function, or actually more accurately of a code block, which I think was related to how jumps and exceptions were set up. But I haven't come across that for AS3, as far as I'm aware there is no limitation on the number of functions or size of each function, perhaps as long as the length fields, offsets etc all fit into 30-bit variables?

The one area I've found a difference between the Flex compiler vs the Falcon2 compiler is around the handling of 'embed' directives, so just checking whether you have any of those?

Otherwise, I would be very interested to try to get to the bottom of this .. I'm wondering whether it may be related to reserved words in some way, but if there are any more details you can provide - we could look at adding extra error outputs into the compiler and try to narrow down on what's happening there.

The inline thing can help depending on what's being done inline of course! But often, it's the resolving of function calls etc which can be a bit more time-consuming, so inlining will help remove that latency. There was an outstanding issue I think regarding the use of inline when using SWC files, so if you are trying that, I'll try to dig out the details on the limitation there..

thanks
